### PR TITLE
dialect: (riscv) Add custom syntax to RdRsImmIntegerOperation

### PR DESCRIPTION
**Pull request by kingiler**
Add custom syntax to RdRsImmIntegerOperation

### CODE REVIEW

File: xdsl/dialects/riscv.py [hunk 1]
```diff
@@ -715,6 +715,23 @@ def __init__(
...
+    def custom_parse_attributes(cls, parser: Parser) -> Mapping[str, Attribute]:
+        attributes = dict[str, Attribute]()
+        attributes["immediate"] = _parse_immediate_value(
+            parser, IntegerType(12, Signedness.SIGNED)
+        )
+        return attributes
+
+    def custom_print_attributes(self, printer: Printer) -> Set[str]:
+        printer.print(", ")
+        match self.immediate:
```
Comment (superlopuh) [inline xdsl/dialects/riscv.py:728]: feels like this could be a helper next to _parse_immediate_value? We'll very likely want to reuse it

**Review by superlopuh — APPROVED** [review]
LGTM

File: xdsl/dialects/riscv.py (hunk 1 above)
Comment (kingiler) [reply xdsl/dialects/riscv.py]: Address in latest commit.

**Review by superlopuh — APPROVED** [review]

File: xdsl/dialects/riscv.py [hunk 2]
```diff
@@ -707,6 +707,19 @@ def __init__(
     def assembly_line_args(self) -> tuple[AssemblyInstructionArg, ...]:
         return self.rd, self.rs1, self.immediate
 
+    @classmethod
+    def custom_parse_attributes(cls, parser: Parser) -> Mapping[str, Attribute]:
+        attributes = dict[str, Attribute]()
+        attributes["immediate"] = _parse_immediate_value(
+            parser, IntegerType(12, Signedness.SIGNED)
```
Comment (adutilleul) [inline xdsl/dialects/riscv.py:714]: what about the child class `RdRsImmShiftOperation` which uses 5 bits immediate ?

**Review by adutilleul — CHANGES_REQUESTED** [review]

File: xdsl/dialects/riscv.py (hunk 2 above)
Comment (kingiler) [reply xdsl/dialects/riscv.py:714]: Address in #1416

**Pull request merged**